src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKFileChooserUI.java

```diff
@@ -980,6 +980,25 @@ protected void sort(Vector<? extends File> v) {
                 v.sort(Comparator.comparing(fsv::getSystemDisplayName));
             }
         }
+
+        @Override
+        public Vector<File> getDirectories() {
+            Vector<File> files = super.getDirectories();
+
+            /*
+             * Delete the "/.." file entry from file chooser directory list in
+             * GTK LAF if current directory is root and files vector contains
+             * "/.." entry.
+             *
+             * It is not possible to go beyond root directory.
+             */
+            File crntDir = getFileChooser().getCurrentDirectory();
+            if (crntDir != null && crntDir.getParentFile() == null &&
+                files.contains(new File("/.."))) {
```

Comment:
Is it not possible to use "FileSystemView.isFileSystemRoot(File)" for this?

Reply:
I have not checked this.
Do you mean to check `FileSystemView.isFileSystemRoot(crntDir )` ?

Reply:
I guess that's what you commented "if current directory is root"

Reply:
Ok... I will check this.

Reply:
> Is it not possible to use "FileSystemView.isFileSystemRoot(File)" for this?

Updated.